docs: Pulled the encoders section of the new shield guide out into its own page (#2510)
zmkfirmware/zmk: docs/docs/features/encoders.md

## Adding Encoder Support

> See the [Encoders](../development/hardware-integration/encoders.md) page for instruction on adding to your shield or modifying existing ones.

Reviewer: ```suggestion
See the [Hardware Integration page for encoders](../development/hardware-integration/encoders.md) for instructions on adding them to your keyboard.
```
